zap/src/main/java/org/zaproxy/zap/extension/search/SearchThread.java

```diff
@@ -270,6 +270,22 @@ private void search() {
                             }
                         }
                     }
+                    if (Type.Tag.equals(reqType)) {
+                        // Tags are based on response so set SearchMatch.Location as resp head
```

Comment:
Tags apply to the whole message not just the response.

Reply:
So it could be set to `null`, I'll have to handle that in the highlighting. But technically a tag could have multiple regexes in different places (req/resp, head/body), and there's no `Alll` location in the enum. (Though I guess I could add one 🤷‍♂️)

Reply:
I guess we could do a combination of the two, take the tag string iterate through the actual tags check which field has a regex and choose the Location from that. If more than one then pass null.

It's more change than I was thinking but probably not horrible.

Reply:
It was about the comment not being accurate, not the lack or incorrect highlight.

Reply:
I don't oppose trying to guess the highlight but we should not ignore tags just because we can't highlight them, that seems wrong and the user will be left wondering why the message is tagged with something and it's still not shown in the search results.

Reply:
Oh, oh well. I removed the comment when I started deriving the location from the tag scanner details.